q2_quality_control/quality_control.py

```diff
@@ -23,30 +20,35 @@ def exclude_seqs(feature_sequences: DNAFASTAFormat,
                  ) -> (pd.Series, pd.Series):
 
     # BLAST query seqs vs. ref db of contaminants (or targets)
-    res = _blast_seqs(
+    hit_ids = _blast_seqs(
         feature_sequences, reference_sequences, evalue=evalue,
         perc_identity=perc_identity, threads=threads,
         perc_query_aligned=perc_query_aligned, method=method)
 
     # convert feature_sequences to series for filtering
     query_series = _dnafastaformats_to_series(feature_sequences)
 
-    # if no hits are in res, early return empty hits and query_series as misses
-    if len(res) < 1:
+    # if no hits are in hit_ids, return empty hits and query_series as misses
+    if len(hit_ids) < 1:
         hits_seqs = pd.Series()
         return hits_seqs, query_series
     # if all query seqs are hits, return query_series as hits and empty misses
-    elif len(res) == len(query_series):
+    elif len(hit_ids) == len(query_series):
         misses_seqs = pd.Series()
         return query_series, misses_seqs
-
-    # filter seqs from seq file
-    res_md = qiime2.Metadata(res)
-    hits_seqs = filter_seqs(query_series, res_md, exclude_ids=False)
-    misses_seqs = filter_seqs(query_series, res_md, exclude_ids=True)
+    # otherwise filter seqs from seq file
+    else:
+        hits_seqs = {}
+        misses_seqs = {}
+        for seq_id, seq in query_series.items():
+            seq = str(seq)
+            if seq_id in hit_ids:
+                hits_seqs[seq_id] = seq
+            else:
+                misses_seqs[seq_id] = seq
 
     # output hits/rejects
-    return hits_seqs, misses_seqs
+    return pd.Series(hits_seqs), pd.Series(misses_seqs)
```

Comment:
This `return` should go in the `else` block too (since the other `return` statements are) - again, just for clarity.